linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/LinkerdAdmin.scala

```diff
@@ -96,6 +112,21 @@ object LinkerdAdmin {
     }
   }
 
+  def extractInterpreterTransformerHandlers(routers: Seq[Router]): Seq[Admin.Handler] = {
```

Comment:
TIOLI: If you end up in flatmap hell, you can always try to resort to for comprehensions. I think in this particular case (stands for the other flatmapped methods as well) you can rewrite this bit of code like that: 

```scala
for {
  r <- routers.collect {case r:WithNameTreeTransformer => r}
  t <- r.transformers.collect {case t:Admin.WithHandlers => t}
  h <- t.adminHandlers
} yield h
```

I think the result will be the same, but I find this one easier to follow, with less nesting.

Reply:
done, collect looks much cleaner than flatmap hell :)